src/linear_algebra/dual.lean

```diff
@@ -558,3 +558,111 @@ end
 end
 
 end subspace
+
+variables {R : Type*} [comm_ring R] {M₁ : Type*} {M₂ : Type*}
+variables [add_comm_group M₁] [module R M₁] [add_comm_group M₂] [module R M₂]
+
+open module
+
+/-- Given a linear map `f : M₁ →ₗ[R] M₂`, `f.dual_map` is the linear map between the dual of
+`M₂` and `M₁` such that it maps the functional `φ` to `φ ∘ f`. -/
+def linear_map.dual_map (f : M₁ →ₗ[R] M₂) : dual R M₂ →ₗ[R] dual R M₁ :=
+linear_map.lcomp R R f
+
+@[simp] lemma linear_map.dual_map_apply (f : M₁ →ₗ[R] M₂) (g : dual R M₂) (x : M₁) :
+  f.dual_map g x = g (f x) :=
+linear_map.lcomp_apply f g x
+
+@[simp] lemma linear_map.dual_map_id :
+  (linear_map.id : M₁ →ₗ[R] M₁).dual_map = linear_map.id :=
+by { ext, refl }
+
+lemma linear_map.dual_map_dual_map (f : M₁ →ₗ[R] M₂) (g : M₂ →ₗ[R] M₁) (φ : dual R M₁) :
```

Comment:
You should have three different `M`s here, not just two.

Reply:
Should we simply show `f.dual_map.comp g.dual_map = (g.comp f).dual_map` instead?

Reply:
Yes, good idea. Perhaps it should be called `dual_map_comp_dual_map` to match.